src/main/java/mate/academy/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,28 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.dao.UserDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Inject
+    private UserDao userDao;
+
+    @Override
+    public User add(User user) {
+        byte[] salt = HashUtil.getSalt();
+        user.setSalt(salt);
```

Comment:
point from common mistake: You should create salt and hash password in `UserService add()` method. move this to appropriate method in `AuthenticationService`  class

Reply:
refactored